scripts/is-merged.js

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env node
+
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { execSync } from 'child_process';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+
+const { branch, merged } = yargs(hideBin(process.argv)).argv;
+
+const isMerged = (branch = 'main', merged) => {
+    let mergedBranched = merged;
+    if (!merged) {
+        const command = execSync('git branch --merged');
+        mergedBranched = command.toString();
+    }
+    const isMerged = mergedBranched.search(branch) > -1;
+    if (isMerged) {
+        console.warn('This branch is up to date.');
+    } else {
+        console.warn('This branch needs to be rebased.');
```

Comment:
If you like, you could try a rebase here too to see if it would rebase cleanly.  Just abort if it requires manual intervention.

Reply:
That could be interesting. Auto history editing could be surprising, on top of everything else, but if it's actually an option, knowing the full possibilities here would allow us to review and construct the right workflow together going forward!

Where I'm struggling more directly with this is in finding a way to trigger this workflow when there are changes on the target branch. Without getting auto run in that context I'm not sure that any work we do here is really _enough_ for it to replace the "This branch is out-of-date with the base branch" functionality that this mimics in hope of preventing the merge commits that come along with it.

Reply:
This Action can auto-rebase: https://github.com/marketplace/actions/auto-rebase